src/components/annotation-widgets/default/MainImage.vue

```diff
@@ -68,25 +82,29 @@ export default class MainImage extends Vue {
     @Prop({required: true, type: String})
     colormapName!: string
     @Prop({required: true, type: Number})
+    zoom!: number
+    @Prop({required: true, type: Number})
     opacity!: number
     @Prop({required: true})
     imageLoaderSettings!: any
     @Prop({required: true, type: Function})
     onImageMove!: Function
-    @Prop({required: true, type: Function})
-    onImageZoom!: Function
+
+    get zoomVal(): number {
+      return this.zoom
+    }
 
     get colorbarDirection(): string {
       return this.colormap[0] == '-' ? 'bottom' : 'top';
     }
 
     saveImage(event: any): void {
-      let node = this.$refs.imageLoader.getContainer();
+      let node = this.$refs.imageLoader.$refs.parent
 
       domtoimage
         .toBlob(node, {
-          width: this.$refs.imageLoader.imageWidth,
-          height: this.$refs.imageLoader.imageHeight
+          width: node.getBoundingClientRect().width,
+          height:  node.getBoundingClientRect().height
         })
```

Comment:
A disadvantage of this change is that now exported images will have wide sections of transparent empty space if the image isn't zoomed in to fill the width of the canvas.

We could mathematically determine how much to export based on the current zoom level, but I'm not sure how important this is. To me, having a good in-page zooming experience is much more important than having the downloaded pngs be 100% perfect. Do you have any opinion @intsco ?